### PR TITLE
Feat: workflow timer and mailer component

### PR DESCRIPTION
**Pull request by Maglitch65**

### CODE REVIEW

File: core/rule-engine/rule-engine-components/pom.xml [hunk 1]
```diff
@@ -71,5 +71,21 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.cronutils</groupId>
+            <artifactId>cron-utils</artifactId>
+            <version>9.2.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.simplejavamail</groupId>
+            <artifactId>simple-java-mail</artifactId>
+            <version>8.12.4</version>
```
Comment (leonside) [inline core/rule-engine/rule-engine-components/pom.xml:87]: Will the version be defined in the parent pom?

File: core/rule-engine/rule-engine-components/src/main/resources/META-INF/services/org/apache/camel/component/email [hunk 2]
```diff
@@ -0,0 +1 @@
+class=com.milesight.beaveriot.rule.components.email.EmailComponent
```
Comment (leonside) [inline core/rule-engine/rule-engine-components/src/main/resources/META-INF/services/org/apache/camel/component/email:1]: Beans do not require the registration of the Camel Component

Comment (Maglitch65) [reply core/rule-engine/rule-engine-components/src/main/resources/META-INF/services/org/apache/camel/component/email]: Ok, I will delete it.

File: core/rule-engine/rule-engine-components/pom.xml (hunk 1 above)
Comment (Maglitch65) [reply core/rule-engine/rule-engine-components/pom.xml]: Done.

**Pull request merged**